packages/compass-components/src/components/guide-cue/guide-cue.tsx

```diff
@@ -0,0 +1,222 @@
+      );
+    };
+    guideCueService.addEventListener('show-cue', listener);
+    return () => {
+      guideCueService.removeEventListener('show-cue', listener);
+    };
+  }, [setIsOpen, cueData, intersectionRef]);
+
+  useEffect(() => {
+    if (!refEl.current) {
+      return;
+    }
+    guideCueService.addCue({
+      ...cueData,
+      isIntersecting: intersectionRef.current,
+    });
+    return () => {
+      guideCueService.removeCue(cueData.cueId, cueData.groupId);
+    };
+  }, [refEl, cueData, intersectionRef]);
+
+  useEffect(() => {
+    // In order to ensure proper positioning, we have introduced
+    // a delay in rendering the cue for a few frames. The issue at
+    // hand is that the LG GC component fails to attach itself
+    // correctly to the refEl element, resulting in an undesired gap
+    // between the GC and refEl. Furthermore, when the position of
+    // refEl changes, the GC does not adjust accordingly. This
+    // somehow seems to be LG issue which we are unable to recreate.
+    return rafraf(() => {
+      setReadyToRender(true);
+    });
+  }, []);
+
+  const onNextGroup = useCallback(() => {
+    setIsOpen(false);
+    guideCueService.markGroupAsVisited(cueData.groupId);
+    guideCueService.onNext();
+  }, [cueData]);
+
+  const onNext = useCallback(() => {
+    setIsOpen(false);
+    guideCueService.markCueAsVisited(cueData.cueId, cueData.groupId);
+    guideCueService.onNext();
+  }, [cueData]);
+
+  useEffect(() => {
+    if (!isOpen || !refEl.current) {
+      return;
+    }
+    const listener = (event: MouseEvent) => {
+      const popover = document.querySelector(
+        `[data-cueid="${getDataCueId(cueData)}"]`
+      );
+      if (!popover) {
+        return;
+      }
+
+      // Clicked within popover
+      if (event.composedPath().includes(popover)) {
+        return;
+      }
+
+      // Clicked within refEl
+      if (event.composedPath().includes(refEl.current!)) {
+        return onNext();
+      }
+
+      // Or else, user clicked outside GC
+      guideCueService.markAllCuesAsVisited();
+      setIsOpen(false);
+    };
+
+    document.addEventListener('mousedown', listener);
+    return () => {
+      document.removeEventListener('mousedown', listener);
+    };
+  }, [isOpen, cueData, refEl, onNext]);
+
+  return (
+    <>
+      {readyToRender && (
+        <LGGuideCue
+          {...restOfCueProps}
+          open={isOpen}
+          numberOfSteps={guideCueService.getCountOfSteps(cueData.groupId)}
+          onDismiss={onNextGroup}
+          onPrimaryButtonClick={onNext}
+          setOpen={() => {
+            // noop
+          }}
+          currentStep={cueData.step || 1}
+          refEl={refEl}
+          data-cueid={getDataCueId(cueData)}
+          tooltipClassName={cx(
+            // Avoid flicker when the component (trigger) becomes invisible
+            !intersectionRef.current && hiddenPopoverStyles
+          )}
+        >
+          {children}
```

Comment:
Extremely small thing, just wondering what you'll think about that. So when adding the GuideCue for the signal popover (and it's actually the same in other places), one thing I found inconvenient is that the component definition looks somewhat like that:


```
cue title

...

trigger body

...

cue description
```

Whereas it seems to me that it would be easier to read if it would be something like that instead:

```
cue title
cue description

...

trigger body
```

I know that this is based on the leafygreen interface here, but maybe we can map their `children` prop to something like `description` prop in our component (and remove `children` prop from the interface)? What do you think? 🤔

Reply:
If it makes our workflow easier, yes totally! I don't have anything against it.